includes/classes/prod_cat_pulldown.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+	class prod_cat_pulldown extends base
+	{
+		var $pulldown, $attributes, $order_by;
+		/**
+		 * @var bool
+		 */
+		private $show_id;
+		/**
+		 * @var array
+		 */
+		private $exclude;
+
+		function __construct(
+			string $pulldown = 'product',
+			array $exclude = [],
+			string $keywords = '',
+			string $order_by = '',
+			bool $attributes = false,
+			$filter_by_option_name = null
+		) {
+
+			$this->exclude = $exclude;
+
+			$this->show_model = true;
+			$this->show_id = false;
+			$this->show_parent = false;
+			$this->show_full_path = false;
+
+			$this->set_selected = 0;
+			$this->values = [];
+
+			$this->keywords = $keywords;
+			$this->order_by = $order_by;
+			$this->option_filter = $filter_by_option_name;
+
+			$this->pulldown = 'product';
+			$this->attributes_join = '';
+
+			if ($attributes) {
+				$this->attributes_join = " LEFT JOIN " . TABLE_PRODUCTS_ATTRIBUTES . " pa USING (products_id)";
+			}
+
+			$this->condition = " WHERE pa.options_id =" . (int)$this->option_filter;
+
+			if ($pulldown == 'category') {
+				$this->pulldown = $pulldown;
+				$this->category_vars();
+			} else {
+				$this->product_vars();
+			}
+
+			$this->run_sql();
```

Comment:
Wouldn't this be better suited at the point of executing `$this->pull_down()`?
There's a few things that appear could benefit by making that move:
- `$this->keyword_search_fields` could be updated/modified to suit,
- `$this->exclude` could be updated/modified (with appropriate handler).
- Ultimately it seems that this class could be pre-defined (instantiated) in page load and not require to be uniquely declared for execution. Would mean that likely some of these parameters may go to the `pull_down` method, but... Then when wanting to use this class only one line of code is needed rather than two (each of which could have problems in it).
- If one wants to obtain differently filtered results, may not have to reinstantiate the variable and display the results.
-

Reply:
great point!  i like it.  feeds into your 2 class suggestion as well.